server/src/main/java/org/elasticsearch/synonyms/SynonymsManagementAPIService.java

```diff
@@ -171,36 +204,39 @@ public void onFailure(Exception e) {
     }
 
     public void getSynonymSetRules(String synonymSetId, int from, int size, ActionListener<PagedResult<SynonymRule>> listener) {
-        client.prepareSearch(SYNONYMS_ALIAS_NAME)
-            .setQuery(QueryBuilders.termQuery(SYNONYMS_SET_FIELD, synonymSetId))
-            .setFrom(from)
-            .setSize(size)
-            .addSort("id", SortOrder.ASC)
-            .setPreference(Preference.LOCAL.type())
-            .setTrackTotalHits(true)
-            .execute(new DelegatingIndexNotFoundActionListener<>(synonymSetId, listener, (l, searchResponse) -> {
-                final long totalSynonymRules = searchResponse.getHits().getTotalHits().value;
-                if (totalSynonymRules == 0) {
-                    l.onFailure(new ResourceNotFoundException("Synonym set [" + synonymSetId + "] not found"));
-                    return;
-                }
-                final SynonymRule[] synonymRules = Arrays.stream(searchResponse.getHits().getHits())
-                    .map(hit -> sourceMapToSynonymRule(hit.getSourceAsMap()))
-                    .toArray(SynonymRule[]::new);
-                l.onResponse(new PagedResult<>(totalSynonymRules, synonymRules));
-            }));
+        checkSynonymSetExists(synonymSetId, listener.delegateFailure((existsListener, response) -> {
```

Comment:
Now we wrap on `checkSynonymSetExists` - which is now faster as it retrieves by ID instead of doing a search.

Reply:
GET request is indeed cheaper than Search, but it is still not free, and I think it is wasteful to do it unnecessary. Should we have this check only after search results return 0 (in the way we do for `deleteSynonymRule`?  

In this and other places.

Reply:
Yes, that's what I meant: if the search results return 0, only after that call `checkSynonymsSetExists`, so we don't need to call it unnecessary. WDYT?

Reply:
got it - changed 👍